yt_dlp/extractor/opencast.py

```diff
@@ -0,0 +1,230 @@
+                                'play_path': m_obj.group('playpath'),
+                                'rtmp_live': True,
+                                'preference': -2,
+                            }
+                        )
+                        extention = m_obj.group('playpath').split(':')
+                        if len(extention) > 1:
+                            track_obj.update({'ext': extention[0]})
+
+                audio_info = track.get('audio')
+                if audio_info is not None:
+                    if 'bitrate' in audio_info:
+                        track_obj.update({'abr': int_or_none(audio_info.get('bitrate'), 1000)})
+                    if 'samplingrate' in audio_info:
+                        track_obj.update({'asr': int_or_none(audio_info.get('samplingrate'))})
+                    audio_encoder = audio_info.get('encoder', {})
+                    if 'type' in audio_encoder:
+                        track_obj.update({'acodec': audio_encoder.get('type')})
+
+                video_info = track.get('video')
+                if video_info is not None:
+                    if 'resolution' in video_info:
+                        track_obj.update({'resolution': video_info.get('resolution')})
+                        resolution = parse_resolution(video_info.get('resolution'))
+                        track_obj.update(resolution)
+                    if 'framerate' in video_info:
+                        track_obj.update({'fps': int_or_none(video_info.get('framerate'))})
+                    if 'bitrate' in video_info:
+                        track_obj.update({'vbr': int_or_none(video_info.get('bitrate'), 1000)})
+                    video_encoder = video_info.get('encoder', {})
+                    if 'type' in video_encoder:
+                        track_obj.update({'vcodec': video_encoder.get('type')})
+
+                formats.append(track_obj)
+
+        self._sort_formats(formats)
+
+        result_obj = {'formats': formats}
+
+        result_obj.update({'id': video_id})
+
+        title = video.get('title')
+        if title is not None:
+            result_obj.update({'title': title})
+
+        series = video.get('seriestitle')
+        if series is not None:
+            result_obj.update({'series': series})
+
+        season_id = video.get('series')
+        if season_id is not None:
+            result_obj.update({'season_id': season_id})
+
+        creator = video.get('creators', {}).get('creator')
+        if creator is not None:
+            result_obj.update({'creator': creator})
+
+        timestamp = parse_iso8601(video.get('start'))
+        if timestamp is not None:
+            result_obj.update({'timestamp': timestamp})
+
+        attachments = video.get('attachments', {}).get('attachment', [])
+        if len(attachments) > 0:
+            thumbnail = attachments[0].get('url')
+            result_obj.update({'thumbnail': thumbnail})
```

Comment:
depending on your intention,

```suggestion
        'thumbnails': [{'url': x} for x in traverse_obj(video, ('attachments', 'attachment', ..., 'url'), default=[])],
```

or

```suggestion
        'thumbnail': traverse_obj(video, ('attachments', 'attachment', ..., 'url'), get_all=False),
```

Also, do not apply as-is. Merge with the above.